src/test/java/com/cloudant/tests/util/Utils.java

```diff
@@ -129,4 +130,31 @@ public static ReplicatorDocument waitForReplicatorToReachStatus(CloudantClient a
     public static String generateUUID() {
         return UUID.randomUUID().toString().replace("-", "");
     }
+
+    /**
+     * When testing against a cluster requests can be routed to different nodes. This means after
+     * a save a document may not appear for all subsequent requests until an internal replication
+     * has finished between the nodes. This method tries to find a document up to maxRetries
+     * times. Even if a document is found it only means it has propagated to the node the request
+     * reached not that it has reached all nodes.
```

Comment:
I think this comment should specify that the retries happen at 0.5s intervals as that's a significant part of why retrying might make it more likely to result in success and helps users realise that it might take a while to return with a high retry count.